defaults/portal2/main.js

```diff
@@ -25,35 +28,46 @@ function wsMessageHandler (event) {
   const data = JSON.parse(event.data);
 
   switch (data.type) {
+    case "authenticated": {
+      ws.send(JSON.stringify({ type: "isGame", value: true }));
+      SendToConsole("clear; echo WebSocket connection established.");
+      return;
+    }
     case "cmd": return SendToConsole(data.value);
     case "ping": return ws.send(JSON.stringify({ type: "pong" }));
+    case "checkMap": {
+      checkmapExpect = data.value;
+      SendToConsole("echo [CheckMapStart];sar_workshop_list;echo [CheckMapEnd]");
+      return;
+    }
+    case "lobby_start" : {
+      menuSession = true;
+      SendToConsole("map " + data.map);
+      return;
+    }
   }
 
 }
 
 // Authenticates and sets up a WebSocket connection
-function wsSetup () {
-
-  // Trying to load a save file prints the user's SteamID to console as a path
-  SendToConsole("load .");
-
-  // Wait for the SteamID to get read from the console
-  onReadSteamID = function (steamid) {
+function wsSetup (token) {
 
-    const protocol = SERVER_ADDRESS.startsWith("https:") ? "wss" : "ws";
-    const hostname = SERVER_ADDRESS.split("://")[1].split("/")[0];
-    const authcode = generateAuthCode();
+  const protocol = SERVER_ADDRESS.startsWith("https:") ? "wss" : "ws";
+  const hostname = SERVER_ADDRESS.split("://")[1].split("/")[0];
 
-    SendToConsole("hideconsole");
-    SendToConsole(`disconnect "Go to ${hostname} and enter the code ${authcode} when prompted."`);
+  if (ws) ws.close();
+  ws = new WebSocket(`${protocol}://${hostname}/api/events/connect`);
```

Comment:
Wouldn't it be possible to just use `SERVER_ADDRESS`, but replace `http` with `ws`? lol

Sliiiiightly shorter implementation

Reply:
I might be stupid